orbotservice/src/main/java/org/torproject/android/service/circumvention/BuiltInBridges.kt

```diff
@@ -0,0 +1,91 @@
+package org.torproject.android.service.circumvention
+
+import android.content.Context
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.json.Json
+import java.io.File
+import java.util.Calendar
+
+@Serializable
+data class BuiltInBridges(
+    @SerialName("meek-azure")
+    var meekAzure: List<Bridge>? = null,
+    var obfs4: List<Bridge>? = null,
+    var snowflake: List<Bridge>? = null,
+    var webtunnel: List<Bridge>? = null,
+) {
+
+    val empty: Boolean
+        get() {
+            return meekAzure.isNullOrEmpty() && obfs4.isNullOrEmpty() &&
+                    snowflake.isNullOrEmpty() && webtunnel.isNullOrEmpty()
+        }
+
+    fun store(context: Context) {
+        getUpdateFile(context).writeText(Json.Default.encodeToString(this))
+    }
+
+
+    companion object {
+
+        const val FILE_NAME = "builtin-bridges.json"
+
+        const val UPDATE_FILE_NAME = "updated-bridges.json"
+
+        private var instance: BuiltInBridges? = null
+
+
+        fun getInstance(context: Context? = null): BuiltInBridges? {
+            if (instance == null && context != null) {
+                try {
+                    instance = read(getUpdateFile(context).readText())
+                }
+                catch (_: Throwable) {}
+            }
+
+            if (instance == null && context != null) {
+                try {
+                    instance =
+                        read(context.assets.open(FILE_NAME).bufferedReader().use { it.readText() })
+                }
+                catch (_: Throwable) {}
+            }
+
+            return instance
+        }
+
+        /**
+         * We consider stored updated-bridges.json file to be outdated after 2 days.
+         */
+        fun isOutdated(context: Context): Boolean {
+            val lastModified = try {
+                getUpdateFile(context).lastModified()
+            }
+            catch (_: Throwable) {
+                0
```

Comment:
Not sure why there's a 0 here. You should log an error to make debugging in this in the future easier incase something ever goes wrong with the call to `getUpdateFile`

Reply:
The only thing this catches is when `lastModified` will throw, because a `SecurityManager` is throwing a `SecurityException`.

That is something that should never happen, and if it does, the only remedy I see is trying to re-load the latest Moat data and hope, we can at least overwrite the file.

I don't see any benefit in logging this, and I'm pretty sure, nobody would ever see such a log entry, anyway.